test/utils/isSameDay_spec.js

```diff
@@ -15,6 +15,15 @@ describe('isSameDay', () => {
     expect(isSameDay(today, tomorrow)).to.equal(false);
   });
 
+  it('returns false for same days of week', () => {
+    // Flags accidentally use of moment's day() function, which returns index
```

Comment:
We use `isSameDay` to mean the same day, period, not the same day of the week. `moment` thinks that date referred to day of month and day referred to day of week which I don't entirely agree with semantically and find confusing (especially since `isSame(foo, 'day')` does what you'd expect in moment). 

The first attempt at improving the speed of this function used `moment`'s `day()` instead of `date()` which was a bug, but that got reverted immediately before a release was even made.

This function has literally been called `isSameDay` for 2 years. I really don't think this PR is the place to update the name if we choose to go that route.